aten/src/ATen/Context.cpp

```diff
@@ -17,9 +17,50 @@
 #if defined(__aarch64__) && !defined(C10_MOBILE)
 #include <cpuinfo.h>
 #endif
-
+#include <iostream>
 namespace at {
 
+static std::vector<std::string> generic_precisions = {"default", "tf32", "bf16"};
```

Comment:
Are there any perf penalties from representing all ops and precisions as strings rather than enumerated types?

Reply:
Thanks for your advice.
Actually, I do not consider the perf penalties here.
I used the strings in python rather than enumerated types for user experience `torch.set_float32_precision(string, ...)` do not require to import some enumerated types.
And in C++ backend, I just think use same strings as python is good enough. They are aligned and readable enough. And we do not need more logical to convert strings (which passed in by user with python) to enumerated types.

Reply:
We often use strings for exactly this reason throughout the PyTorch API. See for example the `driver=` kwarg in `linalg.lstsq`